### PR TITLE
test(atm-protocol): add proptest round-trip coverage

### PR DESCRIPTION
**Pull request by damelLP**
## Summary

Adds property-based round-trip tests to `atm-protocol`. Wire formats are a textbook fit for property testing — the invariant `parse(serialize(x)) == x` catches whole categories of regressions that hand-written tests miss (off-by-one variants, dropped optional fields, encoding quirks). Closes beads-eiu.

- New `crates/atm-protocol/tests/proptest_roundtrip.rs` exercises every `MessageType` (9 variants), every `DaemonMessage` (8 variants), `ClientMessage`, `ProtocolVersion`, and `SessionView` — 5 properties at 256 cases each plus 4 pinned edge-case unit tests.
- `proptest = "1"` added as a dev-dependency on `atm-protocol`.
- Equality is compared via re-serialized `serde_json::Value`, since the protocol types intentionally don't derive `PartialEq` (they wrap free-form `Value` payloads and boxed views).

## What the strategies cover

| Concern | Coverage |
| --- | --- |
| Variant coverage | All 9 `MessageType` + all 8 `DaemonMessage` variants explicitly generated |
| Strings | Empty, ASCII, accents, ZWSP, emoji + combining marks, ~1 KiB strings, arbitrary unicode |
| Numerics | `u64::MAX` for ping/pong seqs, `u16::MAX` for protocol version, full `i64`/`u64` ranges |
| f64 fields | Bounded to ±1e12 — see "Findings" below |
| Optional fields | Both `Some(...)` and `None` exercised across 30 `SessionView` wire fields |
| Nested JSON | Recursive generator for `StatusUpdate` / `HookEvent` payloads, NaN/Inf filtered upfront |

## Findings (worth flagging, not protocol bugs)

**`serde_json` is not 1-ULP exact for extreme-magnitude finite f64s.** First proptest run found `7.302713383307333e-187` round-trips through serialize→parse→serialize as `…334e-187` then `…335e-187` — confirmed via isolated repro outside this crate, the parsed `f64` differs from the original by one ULP. This is an upstream JSON-number formatting limitation, not anything `atm-protocol` does wrong; realistic `cost_usd` / percentage / duration values sit comfortably inside ±1e12, so I bounded the strategy there with an explanatory comment. The original failing seeds are checked in via `proptest_roundtrip.proptest-regressions` so the discovery isn't lost — if anyone later loosens the strategy back to `any::<f64>()`, those seeds will replay and re-trigger the same failure.

**Stack overflow gotcha during development.** Expanding `arb_session_view` to vary all 30 wire fields blew the default 2 MiB test thread stack on debug builds — the cause was monomorphized strategy type complexity, not heap usage (confirmed by `RUST_MIN_STACK=33554432`). Fix: `.boxed()` each themed bundle to type-erase via `dyn Strategy`. The fix is documented inline so future maintainers adding more bundles don't hit the same wall blindly.

## Why this is a separate concern from `parse.rs`

`RawStatusLine`, `RawHookEvent`, etc. are deserialize-only (the daemon ingests Claude Code JSON; it doesn't emit it). They aren't symmetric round-trip candidates so they're correctly excluded from this suite.

## Test plan

- [x] `cargo test -p atm-protocol` — 25 pre-existing tests + 9 new tests, all passing on default stack
- [x] All 17 message variants individually exercised
- [x] Verified the f64 boundary issue is `serde_json`/upstream, not protocol-level (isolated repro)
- [x] Verified `.boxed()` strategy fixes the stack overflow without changing test behavior

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: crates/atm-protocol/tests/proptest_roundtrip.rs [hunk 1]
```diff
@@ -0,0 +1,453 @@
...
+/// f64s — cost in USD, context percentage, duration in seconds — sit
+/// comfortably inside `±1e12`, so this range is the protocol-meaningful
+/// domain for round-trip checking.
+fn arb_realistic_f64() -> impl Strategy<Value = f64> {
+    prop_oneof![
+        Just(0.0_f64),
+        Just(-0.0_f64),
+        Just(1.0_f64),
+        Just(-1.0_f64),
+        -1.0e12_f64..1.0e12_f64,
```
[the remark this answers is not in the document]
Comment (damelLP) [reply crates/atm-protocol/tests/proptest_roundtrip.rs]: Good catch — fixed in 1c3ca32.

You're exactly right: a single `-1e12..1e12` range still admits arbitrarily small magnitudes (the `prop_oneof!` in the original strategy didn't filter out tiny values, just bounded the absolute extremes), so the suite was stable only because uniform sampling of 256 cases is unlikely to land on subnormals. The doc comment was aspirational rather than accurate.

The fix matches your suggestion: pin `0.0` / `-0.0` / `±1.0` as explicit values, then sample positive magnitudes from `[1e-12, 1e12]` and negative magnitudes from `[-1e12, -1e-12]`. Verified stable at 4096 cases per property (~20k generations) with no failures.

Thanks for the review!
